### PR TITLE
Indent workload get and csc get output

### LINKED ISSUE #228: Indent content below each section heading in workload get output to improve readability (possibly also `clustersupplychain get`)
### Description of problem

The volume of content included in the output from `workload get` has increased significantly with several sections and will likely increase further over time.

The section headings aren't currently standing out as headings and parsing the output requires effort.

### Proposed solution

Indent the content below each section heading to make the parent/child relationship visually obvious, enabling viewers to more easily scan the content and focus on the section relevant to their current need.

### Example
**Current output:**
```
tanzu apps workload get app1
---
# app1: Ready
---
Source
type:    source image
image:   gcr.io/pivotal-knative/jc-apps:latest@sha256:5ad9079b0f97a893e8feecd0adf683444cb5c29c8a32e427bef6a67fe019e487

Supply Chain
name:          source-to-url
last update:   19h
ready:         True

RESOURCE          READY   TIME
source-provider   True    19h
deliverable       True    19h
image-builder     True    19h
config-provider   True    19h
app-config        True    19h
config-writer     True    19h

Issues
No issues reported.

Pods
NAME                                     STATUS      RESTARTS   AGE
app1-00001-deployment-58d84d7d5d-2trnp   Running     0          19h
app1-build-1-build-pod                   Succeeded   0          19h
app1-config-writer-5p5df-pod             Succeeded   0          19h

Knative Services
NAME   READY   URL
app1   Ready   http://app1.ns1.127.0.0.1.nip.io

To see logs: "tanzu apps workload tail app1"
```

**Proposed output (3 space indent)**

```
tanzu apps workload get app1
---
# app1: Ready
---
Source
   type:    source image
   image:   gcr.io/pivotal-knative/jc- 
   apps:latest@sha256:5ad9079b0f97a893e8feecd0adf683444cb5c29c8a32e427bef6a67fe019e487

Supply Chain
   name:          source-to-url
   last update:   19h
   ready:         True

   RESOURCE          READY   TIME
   source-provider   True    19h
   deliverable       True    19h
   image-builder     True    19h
   config-provider   True    19h
   app-config        True    19h
   config-writer     True    19h

   Issues
   No issues reported.

Pods
   NAME                                     STATUS      RESTARTS   AGE
   app1-00001-deployment-58d84d7d5d-2trnp   Running     0          19h
   app1-build-1-build-pod                   Succeeded   0          19h
   app1-config-writer-5p5df-pod             Succeeded   0          19h

Knative Services
   NAME   READY   URL
   app1   Ready   http://app1.ns1.127.0.0.1.nip.io

To see logs: "tanzu apps workload tail app1"
```

### Describe alternatives you've considered
Bold / Colorize headings -- I see indenting as a stronger indicator of visual hierarchy. Additionally, we can reserve bold/color to draw attention to important information within the sections (e.g. error status). One negative of the indenting however, is that it increases the character count for each line and increases opportunities for text wrapping.

I'd be interested in other opinions and suggestions.

### PR DESCRIPTION
**Pull request by warango4**
Pull request

### What this PR does / why we need it
Indent get command ouput so each section is visually distinguishable

### Which issue(s) this PR fixes

Fixes #228 

### Describe testing done for PR

- Unit testing
- `workload get` and `csc get` in local cluster

### Additional information or special notes for your reviewer

Signed-off-by: Wendy Arango warango@vmware.com

### CODE REVIEW

**Comment by warango4** [comment]
image

**Comment by warango4** [comment]
image

**Comment by warango4** [comment]
image

**Comment by warango4** [comment]
image

File: pkg/cli-runtime/printer/table/tabwriter.go [hunk 1]
```diff
@@ -34,6 +34,6 @@ const (
 )
 
 // GetNewTabWriter returns a tabwriter that translates tabbed columns in input into properly aligned text.
-func GetNewTabWriter(output io.Writer) *tabwriter.Writer {
-	return tabwriter.NewWriter(output, tabwriterMinWidth, tabwriterWidth, tabwriterPadding, tabwriterPadChar, tabwriterFlags)
+func GetNewTabWriter(output io.Writer, paddingStart bool) *tabwriter.Writer {
```
Comment (shashwathi) [inline pkg/cli-runtime/printer/table/tabwriter.go:37]: Is there any case where we are setting this to false?

Comment (shashwathi) [reply pkg/cli-runtime/printer/table/tabwriter.go:37]: If not, you can keep the func signature `GetNewTabWriter(output io.Writer)` and set the `paddingStart` variable to `false` in func `tabwriter.NewWriter` struct

File: pkg/cli-runtime/printer/tabwriter/tabwriter.go [hunk 2]
```diff
@@ -319,6 +322,10 @@ func (b *Writer) writeLines(pos0 int, line0, line1 int) (pos int) {
 		// if TabIndent is set, use tabs to pad leading empty cells
 		useTabs := b.flags&TabIndent != 0
 
+		if len(line) > 0 && b.paddingStart {
```
Comment (shashwathi) [inline pkg/cli-runtime/printer/tabwriter/tabwriter.go:325]: What is the padding value here? How many spaces padding is done by default and should this be configurable from TabWriter struct?

File: pkg/cli-runtime/printer/table/tabwriter.go (hunk 1 above)
Comment (warango4) [reply pkg/cli-runtime/printer/table/tabwriter.go:37]: Yes, we're actually using it in `False` for `workload list` and `csc list`

Comment (warango4) [reply pkg/cli-runtime/printer/table/tabwriter.go:37]: We're using it in `False` for `csc list` and `wld list` outputs

File: pkg/cli-runtime/printer/tabwriter/tabwriter.go (hunk 2 above)
Comment (warango4) [reply pkg/cli-runtime/printer/tabwriter/tabwriter.go:325]: I'm using the default spaces that are used in the whole table. But we can definitely use another variable for the initial spaces for indentation

Comment (warango4) [reply pkg/cli-runtime/printer/tabwriter/tabwriter.go:325]: [This line](https://github.com/vmware-tanzu/apps-cli-plugin/pull/247/files#diff-cbbb1be8f81b9b35121a45c800ea1decc899ba327a0e8fc36f853808fe3e4164L31) is the one that sets the number of padding spaces at the beginning of the table

File: pkg/cli-runtime/printer/tabwriter/tabwriter.go [hunk 3]
```diff
@@ -319,6 +322,10 @@ func (b *Writer) writeLines(pos0 int, line0, line1 int) (pos int) {
 		// if TabIndent is set, use tabs to pad leading empty cells
 		useTabs := b.flags&TabIndent != 0
 
+		if len(line) > 0 && b.paddingStart > 0 {
```
Comment (shashwathi) [inline pkg/cli-runtime/printer/tabwriter/tabwriter.go:325]: why do you need to check whether paddingStart is above 0 ? If its 0 then there is no padding anyway

**Review by shashwathi — COMMENTED** [review]
Looks good mostly. Minor nit comment

**Review by odinnordico — APPROVED** [review]
LGTM

**Review by shaheerkootteeri — APPROVED** [review]
Looks good to me

**Review by shashwathi — APPROVED** [review]

**Pull request merged**